salt/master.py

```diff
@@ -1309,7 +1309,10 @@ def _file_recv(self, load):
         with salt.utils.fopen(cpath, mode) as fp_:
             if load['loc']:
                 fp_.seek(load['loc'])
-            fp_.write(load['data'])
+            if six.PY3:
+                fp_.write(load['data'].encode())
```

Comment:
You can't just call encode, sometimes it will work, sometime it won't. Use `__salt_system_encoding__`.